contracts/LiquidityBridgeContract.sol

```diff
@@ -367,6 +368,32 @@ contract LiquidityBridgeContract {
         return transferredAmountOrErrorCode;
     }
 
+    function registerPegOut(
+        Quote memory quote,
+        bytes memory signature,
+        uint256 height
+    ) public noReentrancy returns (int256) {
+
+        // check quote.lbcAddress if its valid
+        //todo: should it be validateAndHashQuoteis enough or require(address(this) == quote.lbcAddress, "Wrong LBC address"); ?
+        bytes32 quoteHash = validateAndHashQuote(quote);
+
+        // check if signature is valid
+        require(SignatureValidator.verify(quote.liquidityProviderRskAddress, quoteHash, signature), "Invalid signature");
+
+        // quote.depositHeightLimit is less or equal to the current height
+        require(height <= block.number, "Block height overflown");
+
+        // todo: do we need to registerBridge?
```

Comment:
There should be a transfer of the value sent in the request to `address(this)`.

Reply:
done, made the function payable